internal/hcp/internal/types/link_configuration.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package types
+
+import (
+	"errors"
+
+	"github.com/hashicorp/consul/internal/resource"
+	pbhcp "github.com/hashicorp/consul/proto-public/pbhcp/v1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+var (
+	linkConfigurationNameError = errors.New("Only a single HCP LinkConfiguration resource is allowed and it must be named hcp-link")
+)
+
+func RegisterLinkConfiguration(r resource.Registry) {
+	r.Register(resource.Registration{
+		Type:     pbhcp.LinkConfigurationType,
+		Proto:    &pbhcp.LinkConfiguration{},
+		Scope:    resource.ScopeCluster,
+		Validate: ValidateLinkConfiguration,
+	})
+}
+
+func ValidateLinkConfiguration(res *pbresource.Resource) error {
+	var linkConfig pbhcp.LinkConfiguration
+
+	if err := res.Data.UnmarshalTo(&linkConfig); err != nil {
+		return resource.NewErrDataParse(&linkConfig, err)
+	}
+
+	if res.Id.Name != "hcp-link" {
```

Comment:
Currently we don't support something like an unnamed singleton object so instead we are enforcing that only one name is ever used.